src/rules/typedefWhitespaceRule.ts

```diff
@@ -14,129 +14,102 @@
     }
 
-    public visitVariableDeclarator(node: TypeScript.VariableDeclaratorSyntax): void {
-        this.checkSpace("variable-declarator", node, node.typeAnnotation);
+    public visitSetAccessor(node: ts.AccessorDeclaration) {
+        this.checkSpace("call-signature", node, node.type, node.parameters.end + 1);
+        super.visitSetAccessor(node);
+    }
 
-        super.visitVariableDeclarator(node);
+    public visitVariableDeclaration(node: ts.VariableDeclaration) {
+        this.checkSpace("variable-declaration", node, node.type, node.name.getEnd());
+        super.visitVariableDeclaration(node);
     }
 
-    public checkSpace(option: string, node: TypeScript.ISyntaxElement, typeAnnotation: TypeScript.TypeAnnotationSyntax) : void {
-        if (this.hasOption(option) && typeAnnotation) {
-            var typeAnnotationChildIndex = this.getTypeAnnotationIndex(node);
-            var precedingChild = this.findPrecedingChild(node, typeAnnotationChildIndex);
-            var trailingTrivia = TypeScript.trailingTrivia(precedingChild);
-            var hasLeadingWhitespace = this.hasLeadingWhitespace(trailingTrivia);
+    // TODO: make type of `typeNode` ts.TypeNode | ts.StringLiteral
+    public checkSpace(option: string, node: ts.Node, typeNode: ts.Node, positionBeforeColon: number) {
+        if (this.hasOption(option) && typeNode != null && positionBeforeColon != null) {
+            var hasLeadingWhitespace: boolean;
+            var scanner = ts.createScanner(ts.ScriptTarget.ES5, false, node.getText());
+
+            scanner.setTextPos(positionBeforeColon - node.getStart());
+            hasLeadingWhitespace = scanner.scan() === ts.SyntaxKind.WhitespaceTrivia;
 
             if (hasLeadingWhitespace !== (this.getOption(option) === "space")) {
-                var position = this.positionAfter(precedingChild);
                 var message = "expected " + this.getOption(option) + " in " + option;
-                var failure = this.createFailure(position, 1, message);
-
-                this.addFailure(failure);
+                this.addFailure(this.createFailure(positionBeforeColon, 1, message));
             }
         }
     }
 
-    public hasOption(option: string): boolean {
+    public hasOption(option: string) {
         var allOptions = this.getOptions();
-        if (!allOptions || allOptions.length === 0) {
+        if (allOptions == null || allOptions.length === 0) {
             return false;
         }
 
         var options = allOptions[0];
-
-        if (!options) {
-            return false;
-        }
-
-        return !!options[option];
+        return options == null || options[option] != null;
     }
 
-    private getOption(option: string): string {
+    private getOption(option: string) {
         var allOptions = this.getOptions();
-        if (!allOptions || allOptions.length === 0) {
-            return undefined;
+        if (allOptions == null || allOptions.length === 0) {
+            return;
```

Comment:
I thought we prefer implicit return types? And I could change this to `return null;` instead of `return;` – the main reasoning here is that you never _really_ need to use the `undefined` keyword in TS, and if we can completely remove the use of a particular concept, the language becomes simpler to use.